src/category_theory/derived.lean

```diff
@@ -0,0 +1,179 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.preadditive.projective_resolution
+
+/-!
+# Left-derived functors
+
+We define the left-derived functors `F.left_derived n : C ⥤ D` for any additive functor `F`
+out of a category with projective resolutions.
+
+The definition is
+```
+projective_resolutions C ⋙ F.map_homotopy_category _ ⋙ homotopy_category.homology_functor D _ n
```

Comment:
I wonder whether we should first have a definition for
```suggestion
projective_resolutions C ⋙ F.map_homotopy_category _
```
and make "taking `n`-th" homology a 2nd definition on top of this.

I think that will be useful when we want to compose derived functors, etc...

Reply:
I retract this comment. It only makes sense when we have the actual derived category. But so far we only have the homotopy category. So, we can refactor this later.